### PR TITLE
feat: [C4R-902] count practice programs and ongs with app

### PR DESCRIPTION
**Pull request by OlteanuCosmin**
[C4R-902](https://wearetribus.atlassian.net/browse/C4R-902)

### CODE REVIEW

File: backend/src/modules/_publicAPI/public-api.controller.ts [hunk 1]
```diff
@@ -51,6 +56,19 @@ export class PublicAPIController {
     });
   }
 
+  @Public()
+  @Get('landing-counters')
```
Comment (dragos1195) [inline backend/src/modules/_publicAPI/public-api.controller.ts:60]: Should be able to handle multiple pulling types.

File: backend/src/modules/_publicAPI/public-api.controller.ts [hunk 2]
```diff
@@ -51,6 +56,19 @@ export class PublicAPIController {
...
+  async getLadningCounters(): Promise<{
+    activePracticePrograms: number;
+    ongsWithApplication: number;
+  }> {
+    return {
+      activePracticePrograms: await this.practiceProgramService.countActive(),
```
Comment (dragos1195) [inline backend/src/modules/_publicAPI/public-api.controller.ts:66]: "activePracticePrograms" should be renamed to be generic.

Comment (dragos1195) [inline backend/src/modules/_publicAPI/public-api.controller.ts:66]: Should be in a separate service -> StatisticsService.

File: backend/src/modules/application/services/application.service.ts [hunk 3]
```diff
@@ -669,6 +672,18 @@ export class ApplicationService {
     return true;
   }
 
+  public async countActiveWithCCApplication(): Promise<number> {
+    const count = this.ongApplicationRepository
+      .getQueryBuilder()
+      .leftJoin('application', 'application', 'application.id = application_id')
+      .where('application.pulling_type =:pullingType', {
+        pullingType: ApplicationPullingType.CIVIC_SERVICE,
```
Comment (dragos1195) [inline backend/src/modules/application/services/application.service.ts:680]: In a different service maybe and with pullingType as a parameter.

File: backend/src/modules/organization/controllers/organization-application.controller.ts [hunk 4]
```diff
@@ -23,6 +23,7 @@ import { ApplicationService } from '../../application/services/application.servi
 import { OrganizationApplicationFilterDto } from 'src/modules/application/dto/organization-application.filters.dto';
 import { ApplicationAccess } from 'src/modules/application/interfaces/application-access.interface';
 import { ApplicationRequestService } from 'src/modules/application/services/application-request.service';
+import { Public } from 'src/common/decorators/public.decorator';
```
Comment (dragos1195) [inline backend/src/modules/organization/controllers/organization-application.controller.ts:26]: ?

File: backend/src/modules/practice-program/services/practice-program.service.ts [hunk 5]
```diff
@@ -308,6 +308,12 @@ export class PracticeProgramService {
     });
   }
 
+  public async countActive(): Promise<number> {
```
Comment (dragos1195) [inline backend/src/modules/practice-program/services/practice-program.service.ts:311]: Different service with pullingType as param.

**Review by dragos1195 — CHANGES_REQUESTED** [review]

File: backend/src/modules/_publicAPI/public-api.controller.ts (hunk 2 above)
Comment (birloiflorian) [reply backend/src/modules/_publicAPI/public-api.controller.ts:66]: @dragos1195 daca mutam in statistics service sa ne asiguram ca construim ruta ```statistics/practice-programs```

**Review by birloiflorian — CHANGES_REQUESTED** [review]

File: backend/src/modules/_publicAPI/public-api.controller.ts (hunk 1 above)
Comment (OlteanuCosmin) [reply backend/src/modules/_publicAPI/public-api.controller.ts]: Can handle multiple pulling types.

File: backend/src/modules/_publicAPI/public-api.controller.ts (hunk 2 above)
Comment (OlteanuCosmin) [reply backend/src/modules/_publicAPI/public-api.controller.ts]: Changed to "activeItems".

Comment (OlteanuCosmin) [reply backend/src/modules/_publicAPI/public-api.controller.ts]: Mutat in StatisticsService si redenumit ruta in `statistics/landing-counters`

File: backend/src/modules/application/services/application.service.ts (hunk 3 above)
Comment (OlteanuCosmin) [reply backend/src/modules/application/services/application.service.ts]: Mutat si adaugat pulling type ca parametru.

File: backend/src/modules/_publicAPI/public-api.controller.ts (hunk 1 above)
Comment (OlteanuCosmin) [reply backend/src/modules/_publicAPI/public-api.controller.ts]: Can handle multiple pulling types.

File: backend/src/modules/organization/controllers/organization-application.controller.ts (hunk 4 above)
Comment (OlteanuCosmin) [reply backend/src/modules/organization/controllers/organization-application.controller.ts]: Sters import-ul.

File: backend/src/modules/application/services/application.service.ts [hunk 6]
```diff
@@ -669,6 +672,18 @@ export class ApplicationService {
     return true;
   }
 
+  public async countActiveWithCCApplication(): Promise<number> {
```
[the remark this answers is not in the document]
Comment (OlteanuCosmin) [reply backend/src/modules/application/services/application.service.ts]: Actualizat.

File: backend/src/modules/practice-program/services/practice-program.service.ts (hunk 5 above)
Comment (OlteanuCosmin) [reply backend/src/modules/practice-program/services/practice-program.service.ts:314]: Done.

File: backend/src/modules/application/services/application.service.ts [hunk 7]
```diff
@@ -669,6 +672,20 @@ export class ApplicationService {
     return true;
   }
 
+  public async countActiveWithApplication(
+    pullingType: string,
```
Comment (dragos1195) [inline backend/src/modules/application/services/application.service.ts:676]: Shouldn't be type of an Enum?

File: backend/src/modules/statistics/controllers/statistics.controller.ts [hunk 8]
```diff
@@ -66,4 +68,13 @@ export class StatisticsController {
   ) {
     return this.statisticsService.getOrganizationStatusStatistics(filters);
   }
+
+  @Public()
+  @ApiQuery({ name: 'pulling_type', type: String })
+  @Get('landing-counters')
+  getLandingCounters(
+    @Query('pulling_type') pullingType: string,
```
Comment (dragos1195) [inline backend/src/modules/statistics/controllers/statistics.controller.ts:76]: Type enum

File: backend/src/modules/statistics/services/statistics.service.ts [hunk 9]
```diff
@@ -284,6 +289,24 @@ export class StatisticsService {
     }
   }
 
+  public async getLandingCounters(pullingType: string): Promise<object> {
```
Comment (dragos1195) [inline backend/src/modules/statistics/services/statistics.service.ts:292]: Type enum

File: backend/src/modules/statistics/services/statistics.service.ts [hunk 10]
```diff
@@ -284,6 +289,24 @@ export class StatisticsService {
...
+      await this.applicationService.countActiveWithApplication(pullingType);
+    let activeItems: number;
+    switch (pullingType) {
+      case ApplicationPullingType.PRACTICE_PROGRAM:
+        activeItems = await this.practiceProgramService.countActive();
+        break;
+      case ApplicationPullingType.CIVIC_SERVICE:
+        activeItems = await this.civicCenterService.countActive();
+        break;
+    }
```
Comment (dragos1195) [inline backend/src/modules/statistics/services/statistics.service.ts:303]: Add a default that throws an error

File: backend/src/modules/statistics/services/statistics.service.ts (hunk 9 above)
Comment (dragos1195) [inline backend/src/modules/statistics/services/statistics.service.ts:292]: Return type should be a promise of an interface.

**Review by dragos1195 — CHANGES_REQUESTED** [review]

File: backend/src/modules/application/services/application.service.ts (hunk 7 above)
Comment (OlteanuCosmin) [reply backend/src/modules/application/services/application.service.ts]: Modificat in type Enum.

File: backend/src/modules/statistics/controllers/statistics.controller.ts (hunk 8 above)
Comment (OlteanuCosmin) [reply backend/src/modules/statistics/controllers/statistics.controller.ts]: Modificat.

File: backend/src/modules/statistics/services/statistics.service.ts (hunk 9 above)
Comment (OlteanuCosmin) [reply backend/src/modules/statistics/services/statistics.service.ts]: Modificat.

File: backend/src/modules/statistics/services/statistics.service.ts (hunk 10 above)
Comment (OlteanuCosmin) [reply backend/src/modules/statistics/services/statistics.service.ts:321]: Added default case and try catch block.

File: backend/src/modules/statistics/services/statistics.service.ts (hunk 9 above)
Comment (OlteanuCosmin) [reply backend/src/modules/statistics/services/statistics.service.ts]: Modified.

**Pull request merged**